src/common/tld.js

```diff
@@ -0,0 +1,10 @@
+import { getDomain as getDomain_, getPublicSuffix as getPublicSuffix_ } from 'tldts';
+
+/**
+ * tldts does not respect the public suffix list by default, but can be opt in manually
+ * with the option `allowPrivateDomains`. Hoist the `sharedOpts` can also help avoid
+ * re-creating the object every time.
+ */
+const sharedOpts = { allowPrivateDomains: true };
```

Comment:
Nit. I noticed that you only seem to call `getDomain` and `getPublicSuffix` on inputs which are already valid hostnames (result of `new URL(url).hostname`). In such cases, you can reduce the amount of work tldts performs by default with some extra options (by default tldts will validate hostnames but also check if it's necessary to extract the hostname from input in case it's a URL). This can be disabled with:

```js
const sharedOpts = {
  allowPrivateDomains: true,
  extractHostname: false, // inputs are already hostnames
  validateHostname: false, // inputs are already valid, no need to perform extra validation
  detectIp: false, // in case you know you don't get IPs as inputs you can also get extra speed by disabling this (not sure if that's something you need for your use-case)
};
```

Reply:
Thanks!

Reply:
Wow! I didn't know that before.